src/main/resources/Diagram/DiagramViewSheet.xml

```diff
@@ -67,7 +67,10 @@
     #set ($discard = $xwiki.ssx.use('Diagram.DiagramViewSheet'))
     #set ($discard = $xwiki.jsx.use('Diagram.DiagramViewSheet'))
     {{html clean="false"}}
-    &lt;div class="diagram loading" data-model="$escapetool.xml($tdoc.content)"&gt;&lt;/div&gt;
+
+    ## Check if the query contains the parameter for getting the diagram from URL
+    #set ($urlViewer = ($request.export == 'url'))
+    &lt;div class="diagram loading" #if (!$urlViewer) data-model="$escapetool.xml($tdoc.content)" #end&gt;&lt;/div&gt;
```

Comment:
I would do it the other way around: output the ``data-model`` attribute if the parameter value is empty (i.e. if the parameter is not specified then we assume the diagram source is the content of the current page). My reason for this is because we don't care about the parameter value at this point: it's the viewer JavaScript code that actually looks at the value.